### PR TITLE
People: Handle external contributor check properly

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

As explained by @jsnajdr in https://github.com/Automattic/wp-calypso/commit/2c332d34027d97ec1da28bbe564b4c618fd2ec95#r49403087 there is a bug with how we handle user IDs when checking for contributors in the team edit page: 

> There seems to be a subtle bug here with how `linked_user_ID` is handled. The REST endpoint returns `false` for Jetpack site users that don't have a wpcom mapping. Look up the `linked_user_ID_on_users_endpoints` function in wpcom PHP source to verify that. But here we're looking only for `undefined`. The `??` operator will never be satisfied and `user?.ID` won't be used as fallback.

This PR takes care of that.

#### Testing instructions

@jsnajdr - I'm not so sure about the steps here, but I was hoping you'll know. Generally, knowing that the API can return `false` for the `linked_user_ID`, verifying that the code makes sense and we're covering that case should generally be enough.

### CODE REVIEW

File: client/my-sites/people/edit-team-member-form/edit-user-form.jsx [hunk 1]
```diff
@@ -273,9 +273,11 @@ export default localize(
 	connect(
 		( state, { siteId, user } ) => {
 			const externalContributors = ( siteId && requestExternalContributors( siteId ).data ) || [];
+			const userId = user?.linked_user_ID || user?.ID;
```
Comment (jsnajdr) [inline client/my-sites/people/edit-team-member-form/edit-user-form.jsx]: The `user` prop is guaranteed to be a valid object, and there are places in this component that destructure it without checking. If you don't mind a little drive-by cleanup, all optional chains when accessing `user` can be removed.

**Review by jsnajdr — APPROVED** [review]
Looks good! Thanks for following up.

File: client/my-sites/people/edit-team-member-form/edit-user-form.jsx (hunk 1 above)
Comment (tyxla) [reply client/my-sites/people/edit-team-member-form/edit-user-form.jsx]: Good point, [just noticed it](https://github.com/Automattic/wp-calypso/blob/d64f25956d54275c213f948b3e06180267d73482/client/my-sites/people/edit-team-member-form/index.jsx#L67). 

Happy to clean this up here, this PR is very small anyway.

Comment (tyxla) [reply client/my-sites/people/edit-team-member-form/edit-user-form.jsx]: Addressed in https://github.com/Automattic/wp-calypso/pull/52238/commits/99f8fc8028345f065b3091d6f92dc9c6fa36c232.
